browser/brave_rewards/add_funds_popup.cc

```diff
@@ -0,0 +1,562 @@
+          ContentSettingsPattern::Wildcard();
+      if (secondary)
+        pattern_secondary =
+            ContentSettingsPattern::FromString(std::string(secondary) + "/*");
+
+      map->SetContentSettingCustomScope(pattern, pattern_secondary, type,
+                                        resource_identifier, setting);
+    } else {
+      // For other types use default scope.
+      ContentSetting default_setting =
+          map->GetDefaultContentSetting(type, nullptr);
+      if (current_setting == default_setting)
+        current_setting = CONTENT_SETTING_DEFAULT;
+      map->SetContentSettingDefaultScope(gurl, gurl_secondary, type,
+                                         resource_identifier, setting);
+    }
+  }
+
+  return current_setting;
+}
+
+void PopupContentSettingsBase::ResetContentSetting(HostContentSettingsMap* map,
+                                                   const char* host,
+                                                   ContentSettingsType type,
+                                                   ContentSetting setting) {
+  DCHECK(map);
+  DCHECK(host);
+  DCHECK(type != CONTENT_SETTINGS_TYPE_PLUGINS);
+
+  const GURL gurl(host);
+  if (setting == CONTENT_SETTING_DEFAULT ||
+      setting != map->GetContentSetting(gurl, GURL(), type, std::string())) {
+    map->SetContentSettingDefaultScope(gurl, GURL(), type, std::string(),
+                                       setting);
+  }
+}
+
+}  // namespace
+
+namespace brave_rewards {
+
+// Pass-through to PopupContentSettingsBase which is in an anonymous namespace.
+class AddFundsPopupContentSettings : public PopupContentSettingsBase {
+ public:
+  AddFundsPopupContentSettings(Profile* profile)
+      : PopupContentSettingsBase(profile) {}
+  DISALLOW_COPY_AND_ASSIGN(AddFundsPopupContentSettings);
+};
+
+AddFundsPopup::AddFundsPopup()
+    : add_funds_popup_(nullptr),
+      popup_content_settings_(nullptr),
+      rewards_service_(nullptr) {}
+
+AddFundsPopup::~AddFundsPopup() {
+  ClosePopup();
+}
+
+// Show existing or open a new popup.
+void AddFundsPopup::ShowPopup(content::WebContents* initiator,
+                              RewardsService* rewards_service) {
+  add_funds_popup_ ? Focus() : OpenPopup(initiator, rewards_service);
+}
+
+void AddFundsPopup::OpenPopup(content::WebContents* initiator,
+                              RewardsService* rewards_service) {
+  DCHECK(!add_funds_popup_);
+  DCHECK(initiator);
+  DCHECK(rewards_service);
+  if (!initiator || !rewards_service)
+    return;
+
+  const std::map<std::string, std::string> addresses =
+      rewards_service->GetAddresses();
+  if (addresses.empty())
+    return;
+
+  content::WebContentsDelegate* wc_delegate = initiator->GetDelegate();
+  if (!wc_delegate)
+    return;
+
+  const GURL gurl(kAddFundsUrl);
+  const GURL referrer_gurl(std::string(brave_toolbar::kInternalUIScheme) +
+                           kRewardsHost);
+  const content::Referrer referrer(referrer_gurl,
```

Comment:
is it necessary to set the referrer here at all? just wondering

Reply:
No, it can be an empty referrer.